tests/connect_two.rs

```diff
@@ -41,6 +41,34 @@ async fn connect_two_nodes_by_peer_id() {
         .expect("should've connected");
 }
 
+// Ensure that duplicate connection attempts don't cause hangs.
+#[async_std::test]
+async fn connect_duplicate_targets() {
+    tracing_subscriber::fmt::init();
+
+    let node_a = Node::new("a").await;
+    let node_b = Node::new("b").await;
+
+    let (b_key, mut b_addrs) = node_b.identity().await.unwrap();
+    let b_id = b_key.into_peer_id();
+    let b_addr = b_addrs.pop().unwrap();
+
+    // test duplicate connections by address
+    for _ in 0..3 {
+        let _ = timeout(Duration::from_secs(1), node_a.connect(b_addr.clone()))
+            .await
+            .unwrap();
```

Comment:
Why is it ok to ignore this `Result<(), anyhow>`?

Reply:
It's sufficient to verify that there is no hang, regardless of what the connection attempt returns.